### DISCUSSION TITLE
SSL certificate problem: certificate has expiredError

### DISCUSSION THREAD

**Discussion by MikleBas** [magicbug/Cloudlog#1784] [Q&A]
I noticed that the command from the cron does not work
0 */6 * * * curl --silent https://<URL-To-Cloudlog>/index.php/clublog/upload/<username-with-clublog-login> &>/dev/null
All certificates on my server are up-to-date and fresh, updated automatically all the time.

I ran a command to the console according to my data in verbose mode
curl -v https://my-site.com/index.php/clublog/upload/my_ham_call

Where is the mistake?

**Comment by phl0** [comment]
> I ran a command to the console according to my data in verbose mode curl -v https://my-site.com/index.php/clublog/upload/my_ham_call

And what was the output of that?

**Reply by MikleBas** [reply]
in console server output 
SSL certificate problem: certificate has expiredError

**Reply by magicbug** [reply]
This sounds like a server issue rather than a Cloudlog one, without seeing logs and the SSL cert its impossible to offer a suggestion

**Reply by MikleBas** [reply]
1. I do not have a **clublog** folder in the specified path
https://my-site.com/index.php/clublog/upload/my_ham_call
2. There are no folders and no files in the https://my-site.com/upload/ folder
3. If you follow your developer's logic, then in the upload folder there should be a directory with my callsign ex. my_call

Probably the correct permissions on the folders were not set during installation? Then what data should be on the path https://my-site.com/index.php/clublog/upload/my_ham_call ??? Where can they get it from?

**Reply by phl0** [reply]
> in console server output SSL certificate problem: certificate has expiredError

Then opening the Cloudlog istance itself should also lead to an error (if you didn't permanently accept the exception). Are you using official certificates or self-signed ones?

**Reply by MikleBas** [reply]
1. Certificate from Let's Encrypt
2. Trying to go along the path curl -v https://MY-SITE/upload/MY-CALL
there is no cerficate error in the console now :)
But there is a natural server response 404 :)
3. Now the question is - what data should be in the upload package??

**Reply by phl0** [reply]
>     2. Trying to go along the path curl -v https://MY-SITE/upload/MY-CALL
>        there is no cerficate error in the console now :)
>        But there is a natural server response 404 :)

Sure because it should be https://MY-SITE/index.php/upload/MY-CALL

>     3. Now the question is - what data should be in the upload package??

Nothing. `upload` is not referring to a folder here but to a function.

**Reply by MikleBas** [reply]
Wonderful!
In this case, when you click on the link https://MY-SITE/index.php/upload/MY-CALL
server returns 404 error
With your permission, I repeat the question - what should I see when I click on the link https://MY-SITE/index.php/upload/MY-CALL ???

**Reply by phl0** [reply]
> Wonderful! In this case, when you click on the link https://MY-SITE/index.php/upload/MY-CALL server returns 404 error With your permission, I repeat the question - what should I see when I click on the link https://MY-SITE/index.php/upload/MY-CALL ???

Well in the ideal case just nothing. It should output errors in case clublog is not reachable. For example:

![Screenshot from 2022-11-15 16-09-18](https://user-images.githubusercontent.com/7112907/201954224-04d55b0c-2d85-45e3-93a7-fc67b333e751.png)

But to be honest: 404 errors do have nothing to do with SSL certificates basically (if you do not redirect SSL traffic to your webpage to a different www folder). That are two different issues I would say.

**Reply by MikleBas** [reply]
Ok, let's freeze the problem with the ssl certificate for now!
What should I do now with uploading my data to your server?
When I go to the site https://clublog.org/ in my personal account, I do not see fresh data from the connection log, only those that I imported manually

**Reply by phl0** [reply]
QRX. @magicbug is Cloublog upload currently working at all? I remember that there was an issue ...

**Reply by AndreasK79** [reply]
Which version? For v2, no call is required. It was rewritten to fetch all with credentials and upload.

**Reply by MikleBas** [reply]
Yes, the second version

**Reply by AndreasK79** [reply]
And did you try without the call?

**Reply by MikleBas** [reply]
![2](https://user-images.githubusercontent.com/99086477/201963141-9dfa9964-6295-4667-9506-39c60bcf38e3.png)

Error 404

**Reply by AndreasK79** [reply]
index.php/clublog/upload

**Reply by MikleBas** [reply]
The full path is
https://my-site.com/index.php/upload
The result is in the screenshot above.

**Reply by phl0** [reply]
> The full path is https://my-site.com/index.php/upload The result is in the screenshot above.

Should be `index.php/clublog/upload`.

**Reply by AndreasK79** [reply]
Yes, I read the full path, but where is clublog in the path?

It is index.php/clublog/upload, not index.php/upload.

Your picture says get /index.php/upload/

**Reply by MikleBas** [reply]
![3](https://user-images.githubusercontent.com/99086477/201965813-b11ed061-54d0-47fb-a054-b23634fb45e6.png)

**Reply by MikleBas** [reply]
> Yes, I read the full path, but where is clublog in the path?
> 
> It is index.php/clublog/upload, not index.php/upload.
> 
> Your picture says get /index.php/upload/

![4](https://user-images.githubusercontent.com/99086477/201966748-d73d14c1-4ce8-4a15-a02f-04a1ad0e61a5.png)

**Here we come to where the conversation started :)
Certificate error**

**Reply by AndreasK79** [reply]
Thanks, at least we have excluded one possible error.

**Reply by phl0** [reply]
> **Here we come to where the conversation started :) Certificate error**

And that is an SSL problem instead of Cloudlog :) curl is probably stumbling across the same issue that the certificate is expired. Now confirmed by browser :)

**Reply by MikleBas** [reply]
> > **Here we come to where the conversation started :) Certificate error**
> 
> And that is an SSL problem instead of Cloudlog :) curl is probably stumbling across the same issue that the certificate is expired. Now confirmed by browser :)

**Reply by MikleBas** [reply]
where can I privately send the address of the site on which the program is running?

**Reply by phl0** [reply]
> where can I privately send the address of the site on which the program is running?

I you want me to test, check my email on qrz.

**Reply by phl0** [reply]
Got it. The SSL certificate on that is valid on my end.... weird

**Reply by phl0** [reply]
I had a deeper look. The SSL certificate presented under the URL you mentioned is valid. The thing is: the certificate for the clublog upload itself from your Cloudlog instance is regarded as expired. So the SSL certificate we are talking about is the one of clublog. The PHP code cannot connect to clublog and throws that error message. 

This will probably be not easy to debug. I would suggest to try a `curl -v "https://clublog.org/putlogs.php"` from your server machine and see what it says about the SSL certificate.

**Reply by phl0** [reply]
Here is the output of my machine for reference:

![Screenshot from 2022-11-16 00-37-12](https://user-images.githubusercontent.com/7112907/202047585-d39b7d32-5e4c-4261-aea2-0d9648b6b3ef.png)

**Reply by MikleBas** [reply]
Everything seems to be fine, I don't see any errors.

![1](https://user-images.githubusercontent.com/99086477/202150248-458820cb-eb83-4ea4-a5ba-86fc8d2ef622.png)

**Reply by MikleBas** [reply]
It looks like clublog has turned on some sort of blocking by country, and now it does not allow access to itself through the browser,
and does not allow you to connect through the console

` * TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200
< server: nginx
< date: Wed, 16 Nov 2022 10:06:39 GMT
< content-type: text/html; charset=UTF-8
< vary: Accept-Encoding
< set-cookie: ci_cloudlog=7t64uhuafrtgniv2f61d4gi848v740sb; path=/; HttpOnly
< expires: Thu, 19 Nov 1981 08:52:00 GMT
< cache-control: no-store, no-cache, must-revalidate
< pragma: no-cache
< strict-transport-security: max-age=31536000
<
* Connection #0 to host MY-SITE left intact
Failed to connect to clublog.org port 443: Connection refusedError`
